ddspipe_yaml/include/ddspipe_yaml/YamlValidator.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <ddspipe_yaml/library/library_dll.h>
+#include <ddspipe_yaml/Yaml.hpp>
+#include <ddspipe_yaml/YamlReader.hpp>
+
+namespace eprosima {
+namespace ddspipe {
+namespace yaml {
+
+// TODO
+class YamlValidator
```

Comment:
The class YamlReader already deals with some errors related to parsing the YAML configuration, why creating a new class?

Reply:
How does the `YamlReader` deal with some errors related to parsing? After looking at it for a bit, I guess you're referring to how the `YamlReader` implicitly detects if there are missing mandatory tags? I don't know. It feels like asking for too much from the `YamlReader` to load the validation functionality into it. The `YamlValidator` has growth potential. I would keep them separate.